src/Extensions/AddToCampaignExtension.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace SilverStripe\CampaignAdmin\Extensions;
+
+use RuntimeException;
+use SilverStripe\AssetAdmin\Controller\AssetAdmin;
+use SilverStripe\CampaignAdmin\AddToCampaignHandler;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\ArrayLib;
+use SilverStripe\Core\Extension;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Extension applied to a Controller to add the ability to add items to a campaign.
+ * This extension will only be applied if the `campaign-admin` module is installed.
+ *
+ * @extends Extension<AssetAdmin>
+ */
+class AddToCampaignExtension extends Extension
+{
+    private static array $allowed_actions = [
+        'addToCampaignForm',
+    ];
+
+    /**
+     * Data class for managed data objects (e.g. pages, files) that are managed by the campaign admin.
+     * This must be set via config elsewhere i.e. yml config or else a RuntimeException will be thrown
+     */
+    private static ?string $campaign_admin_managed_data_class = null;
```

Comment:
Can you please explain why setting it here doesn't work? That sounds a bit weird, almost like there's some underlying bug in the config system or something, but I assume it's much more mundane than that.

Reply:
I also can only see where it's set when `asset-admin` and `cms` are installed, implying that there will always be a runtime exception if those modules aren't installed?

Reply:
Have updated the comment to clarify